### ISSUE TITLE
Error in if (traitType == «binary») { : argument is of length zero

### ISSUE THREAD

**Issue by sboer** [weizhouUMICH/SAIGE#27]
Hi!

I have been running SAIGE version 0.20 without problems, but after upgrading to version 0.26 I keep getting an error when trying to run step 2. 
The cmd.sh example script works, so I guess the installation is ok. 

Step 1 run seems to run without problem on this particular phenotype, and regarding the trait type, this is in the log-file:

$traitType
[1] «binary»
 -and-
Migrene  is a binary trait

When running step 2, I keep getting this error message:

Error in try(if (length(which(opt == ««)) > 0) stop(«Missing arguments»)) : 
  Missing arguments
17756  samples have been used to fit the glmm null model
variance Ratio is  0.906 
58173  sample IDs are found in sample file
[1] 58173     4
[1] «IID»          «IndexInModel» «IndexDose.x»  «IndexDose.y» 
17756  samples were used in fitting the NULL glmm model and are found in sample file
minMAC:  1 
minMAF:  1e-04 
Minimum MAF of markers to be testd is  1e-04 
Analysis started at  1.52e+09 Seconds
setgenoTest here

file extenstion is dose
set GenoTestObj from dosage file!
NSampleTest 58173
Mtest 2928

Testing markers: 2928 , samples: 58173
isVariant:  TRUE 
Error in if (traitType == «binary») { : argument is of length zero
Calls: SPAGMMATtest
Execution halted

This is the code from step 1:

logfile_step1=./Migraine_women_step1.log
{
 Rscript /mnt/work/Programs/SAIGE_0.26/SAIGE/extdata/step1_fitNULLGLMM.R
--plinkfile=/path/Dataset_pruned \
--phenoFile=/path/Headache_women_genotyped.txt \
--phenoCol=Migrene \
--covarColList=PC1,PC2,PC3,PC4,Age \
--sampleIDColinphenoFile=IID \
--traitType=binary \
--skipModelFitting=FALSE \
--outputPrefix=Migraine_women
} 2>&1 | tee «$logfile_step1»¨

And this is the code from step 2

logfile_step2=./Migrene_women_step2.log
{
Rscript /mnt/work/Programs/SAIGE_0.26/SAIGE/extdata/step2_SPAtests.R \
--dosageFile=/path/Dataset_imputed.dose \
--dosageFileNrowSkip=0 \
--dosageFileNcolSkip=5 \
--dosageFilecolnamesSkip=CHR,SNP,POS,A1,A2 \
--minMAF=0.0001 \
--sampleFile=/path/samplelist_IID.txt \
--GMMATmodelFile=./Migraine_women.rda \
--varianceRatioFile=./Migraine_women.varianceRatio.txt \
--SAIGEOutputFile=./Migraine_women_results.txt \
--numLinesOutput=2 \
--IsOutputAFinCaseCtrl=TRUE 
} 2>&1 | tee «$logfile_step2»

I have tried several phenotypes and different dosage-files, but I keep getting the same error message.. 

Could you please help me figure out what I am doing wrong here..? 

Thanks,
Sigrid

**Comment by weizhouUMICH** [comment]
Hi Sigrid, 

In version 0.26 (version >= 0.23), some redundant arguments that have already been specified in step 1 are not required in step 2 and they are stored in the step 1 output. Therefore, step 2 in version 0.26 assumes the step1 output contains the traitType information. You may want to re-run step 1 using version 0.26. 

Thanks,
Wei

**Comment by sboer** [comment]
Hi!

Thanks for answering my question!

I tried to run on the "old" step1-files first, but as it didn´t work I ran it again with the new version of SAIGE. Could it somehow have managed to run the "old" SAIGE-version, even if I specified the new R-script in the command? I did have some trouble initially when installing the new version, but it seemed to work out, and step 1 was running without trouble. 

I will try to run it again with the 26.6-version, and try to specify another path to see if it works.. 

Sigrid

**Comment by sboer** [comment]
Ah, it looks like there must have been an issue with the installation afterall - sorry! I did run step 1 with the new version, but it must have looked up a script from the old version anyway. I install the 0.26.6 version and made sure everything was up to date, and then it worked. 

I have also struggeled with the "error: inv_sympd(): matrix is singular or not positive definite", in some cases, most phenotypes could be run due to this error. But with the 0.26.6 version it worked!

Thanks!

Sigrid
